File: orm/src/main/java/com/example/orm/PersonRepository.java
package com.example.orm;

import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase;

import javax.enterprise.context.ApplicationScoped;
import javax.transaction.Transactional;
import java.util.List;

@ApplicationScoped
@Transactional
public class PersonRepository implements PanacheRepositoryBase<Person,Integer> {

    public Person findByName(String name){
        return find("name", name).firstResult();
    }

    public Person findPersonById(Integer id){
        return find("id", id).firstResult();
    }

    public List<Person> findAlive(){
        return list("status", Status.Alive);
    }

    public void deleteStefs(){
        delete("name", "Stef");
    }

    public List<Person> findAllPerson() { return listAll(); }

}

File: kotlin-orm/src/main/kotlin/com/example/orm/PersonRepository.kt
package com.example.orm

import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase
import javax.enterprise.context.ApplicationScoped
import javax.transaction.Transactional

@ApplicationScoped
@Transactional
open class PersonRepository(): PanacheRepositoryBase<Person, Int> {
    open fun findByName(name: String): Person {
        return find("name", name).firstResult()
    }

    open fun findPersonById(id: Int?): Person? {
        return find("id", id).firstResult()
    }

    open fun findAlive(): List<Person>? {
        return list("status", Status.Alive)
    }

    open fun deleteStefs() {
        delete("name", "Stef")
    }

    open fun deleteById(id: Int?) {
        delete("id", id)
    }

    open fun findAllPerson(): List<Person>? {
        return listAll()
    }
}